### PR TITLE
Add change in Go to customize to which channel message will be passed

### PR DESCRIPTION
**Pull request by ezavgorodniy**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -111,6 +111,12 @@ fabric.properties
 .idea/**/markdown-navigator.xml
 .idea/**/markdown-navigator/
 
+# GoLand files
+.idea/.gitignore
+.idea/kube-job-notifier.iml
+.idea/modules.xml
+.idea/vcs.xml
```
Comment (errm) [inline .gitignore:118]: I don't think stuff specific to your own environment belong in this file, normally you can ignore these kind of files in a global gitignore!

File: Dockerfile [hunk 2]
```diff
@@ -4,7 +4,7 @@ COPY . /go/src/app
 RUN go build -o ./kube-job-notifier *.go
 
 FROM alpine:3.16
-LABEL maintainer="yutachaos <bumplive@gmail.com>"
+LABEL maintainer="yevhen <yevhen.zavhorodnii@gmail.com>"
```
Comment (errm) [inline Dockerfile]: a) You should possibly use your cookpad.com email address, and not a gmail one
b) I don't think we want this change going upstream so perhaps it should be in a different PR or something!

File: controller.go [hunk 3]
```diff
@@ -31,6 +31,7 @@ const (
 	controllerAgentName = "cronjob-controller"
 	intTrue             = 1
 	searchLabel         = "controller-uid"
+	slackAnnotationName = "slack-channel"
```
Comment (errm) [inline controller.go:34]: It seems like common practice to namespace annotations to the tool that is expected to read them.
```suggestion
	slackAnnotationName = "kube-job-notifier/slack-channel"
```

File: controller.go [hunk 4]
```diff
@@ -94,7 +95,11 @@ func NewController(
 				StartTime:   newJob.Status.StartTime,
 			}
 			for name, n := range notifications {
-				err := n.NotifyStart(messageParam)
+				if cronJob == "" {
```
Comment (errm) [inline controller.go:98]: I don't think we should do this since it breaks the previous behaviour of this tool.

How about looking at an annotation like `kube-job-notifier/notify`

If this is not present or looks like `kube-job-notifier/notify = start,success,failed` then the current behaviour is preserved, if you only wanted to notify on sucess you could set `kube-job-notify/notify = failed`

another alternative could be three annotations like `kube-job-notifier/start`, `kube-job-notifier/success` and `kube-job-notifier/failed` and these could be set to `notify` or `ignore` to override the default behaviour from the global environment variables!

File: controller.go (hunk 3 above)
Comment (errm) [reply controller.go:34]: Also as existing behaviour is to have a different channel per success or failure perhaps it would be good to have a way to set different channels per job.

e.g.

`kube-job-notifier/slack-channel-success` or `kube-job-notifier/slack-channel-start` or `kube-job-notifier/slack-channel-failed`

File: pkg/notification/slack.go [hunk 5]
```diff
@@ -54,15 +54,14 @@ func newSlack() slack {
...
 }
 
-func (s slack) NotifyStart(messageParam MessageTemplateParam) (err error) {
+func (s slack) NotifyStart(messageParam MessageTemplateParam, slackChannel string) (err error) {
 
 	if !isNotifyFromEnv("SLACK_STARTED_NOTIFY") {
 		return nil
 	}
 
-	succeedChannel := os.Getenv("SLACK_SUCCEED_CHANNEL")
```
Comment (errm) [inline pkg/notification/slack.go:63]: Shouldn't we fall back to getting the channel from the environment var if the annotation wasn't set? We don't want to change the behaviour if we can help it!

File: pkg/notification/slack.go [hunk 6]
```diff
@@ -97,15 +96,14 @@ func getSlackMessage(messageParam MessageTemplateParam) (slackMessage string, er
...
 }
 
-func (s slack) NotifySuccess(messageParam MessageTemplateParam) (err error) {
+func (s slack) NotifySuccess(messageParam MessageTemplateParam, slackChannel string) (err error) {
 
 	if !isNotifyFromEnv("SLACK_SUCCEEDED_NOTIFY") {
 		return nil
 	}
 
-	succeedChannel := os.Getenv("SLACK_SUCCEED_CHANNEL")
```
Comment (errm) [inline pkg/notification/slack.go:106]: Ditto!

File: pkg/notification/slack.go [hunk 7]
```diff
@@ -136,15 +134,14 @@ func (s slack) NotifySuccess(messageParam MessageTemplateParam) (err error) {
...
 }
 
-func (s slack) NotifyFailed(messageParam MessageTemplateParam) (err error) {
+func (s slack) NotifyFailed(messageParam MessageTemplateParam, slackChannel string) (err error) {
 
 	if !isNotifyFromEnv("SLACK_FAILED_NOTIFY") {
 		return nil
 	}
 
-	failedChannel := os.Getenv("SLACK_FAILED_CHANNEL")
```
Comment (errm) [inline pkg/notification/slack.go:145]: Ditto!

**Review by errm — CHANGES_REQUESTED** [review]

File: charts/kube-job-notifier/Chart.yaml [hunk 8]
```diff
@@ -24,5 +24,5 @@ version: 0.1.6
 appVersion: "v0.0.21"
```
Comment (jportasa) [inline charts/kube-job-notifier/Chart.yaml:24]: if you change something in the image **appVersion**, if you change something in the chart: **version**

File: controller.go (hunk 3 above)
Comment (jportasa) [inline controller.go:34]: Don't you add annotations variable in the values.yaml?
